nasa/harmony
# Harmony 1897

## Jira Issue ID
HARMONY-1897

## Description
Adds support for adding labels to jobs as well as removing labels from jobs.

## Local Test Steps
1. Run the migrations
2. Verify that any previous labels you added to jobs still show up in the job status
3. Add labels to a job:
```
curl  -XPUSH http://localhost:3000/labels -d '{"jobID": ["<YOUR JOB ID>"], "label": ["foo", "bar", "foo", "buzz"]}'  -H "content-type: application/json" -H "Authorization: bearer $MY_UAT_TOKEN"
```
4. Verify that the labels are added using the workflow-ui or job status
5. Try some other labels with leading/trailing spaces and different casing
6. Delete labels from the job
```
curl  -XDELETE http://localhost:3000/labels -d '{"jobID": ["<YOUR JOB ID>"], "label": ["foo", "bar"]}'  -H "content-type: application/json" -H "Authorization: bearer $MY_UAT_TOKEN"
```
7. Verify that the labels are removed using the workflow-ui or job status
8. Read the documentation for labels in the online documentation page

## PR Acceptance Checklist
